### DISCUSSION TITLE
How should we acces to nuxt context on Nuxt3

### DISCUSSION THREAD

**Discussion by david8z** [vuejs/pinia#1109] [Help and Questions] [answered]
When accesing nuxt context for example `this.$nuxt.route.params` on Nuxt3 I get the follwing warnings

```
Accessing __v_isRef is not supported in Nuxt3.

Accessing __v_isReadonly is not supported in Nuxt3.
```

Is there a different recommended way?

**Comment by Lanayx** [comment]
I'm getting the same warnings when using composables for example
`const event = useRequestEvent(this.$nuxt)`
however it still works even with warnings

**Comment by bros4president** [comment]
Same here! Is this a Nuxt issue or Pinia?

**Comment by huangbh1024** [comment]
i also have this promble ,how to resolve it

**Comment by HendrikJan** [comment]
To get route information in Nuxt3 you can use:

```javascript
const route = useRoute();
const router = useRouter();
```

**Comment by marianheinsen** [comment]
Same problem for me

**Comment by nikkistorme** [comment]
I'm having the same issue. I want to use Nuxt context from inside a Pinia store so I can access the `useSupabaseClient` composable, but I'm getting the same errors @david8z mentioned above.

My action:
```
    async signIn(credentials) {
      this.loading = true;
      const supabase = this.$nuxt.useSupabaseClient();
      await supabase.auth.signIn({
        email: credentials.value.email,
        password: credentials.value.password,
      });
      this.loading = false;
    }
```

Console message:
```
Accessing __v_isRef is not supported in Nuxt3.
Accessing __v_isReadonly is not supported in Nuxt3.
Accessing useSupabaseClient is not supported in Nuxt3.
```

**Comment by LinusBorg** [comment]
This error is coming from nuxt, not pinia. Here's an example of an older issue where some other 3rd party lib triggered this in Nuxt3, and it was fixed in Nuxt. So I'd assume it's also something to fix on their end.

**Comment by T-Zahil** [comment]
Any news about this? We still can't access the context from Pinia despite what's explained in the docs 😕

**Comment by bros4president** [comment, marked as answer]
@T-Zahil and everyone else reaching this thread:

[From @pinia/nuxt 0.3.0 onwards](https://github.com/vuejs/pinia/blob/b6382452030a3b006c5bf61302e699f19d4d88e5/packages/nuxt/CHANGELOG.md#030-2022-07-13), you can access nuxt with `useNuxtApp()`, globally in a store.

So you just need to replace `this.$nuxt.foo` with `useNuxtApp().foo` (or assign useNuxtApp output to a variable of course).

This is due to Nuxt [dropping legacy context in rc.5](https://github.com/nuxt/framework/releases/tag/v3.0.0-rc.5).

Thanks to the team for taking care of this caveat. Logs are way cleaner now without the warnings :)!
